### PR TITLE
add osousername to tenants db

### PR DESCRIPTION
**Pull request by surajssd**
Closes https://github.com/openshiftio/openshift.io/issues/2635

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @surajssd for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin - for example exclude certain file types so if PR contains only them it won't be checked.

**Comment by surajssd** [comment]
Removed few tests that I did add to test the migrations, was using code from wit, once we have that wit code abstracted out into separate repo i think we can add those tests!

File: controller/tenant.go [hunk 1]
```diff
@@ -100,7 +100,11 @@ func (c *TenantController) Setup(ctx *app.SetupTenantContext) error {
 
 	// create openshift config
 	openshiftConfig := openshift.NewConfig(c.defaultOpenshiftConfig, user, cluster.User, cluster.Token, cluster.APIURL)
-	tenant := &tenant.Tenant{ID: ttoken.Subject(), Email: ttoken.Email()}
+	tenant := &tenant.Tenant{
+		ID:          ttoken.Subject(),
+		Email:       ttoken.Email(),
+		OsoUsername: openshiftUsername,
```
Comment (aslakknutsen) [inline controller/tenant.go]: OSUsername is more correct. Oso assumes openshift online which is wrong

**Comment by surajssd** [comment]
@aslakknutsen I have made changes as you requested :-)

**Comment by aslakknutsen** [comment]
@surajssd This won't work in current form. You update the datastore and change PATCH/DELETE to require the new data without any migration. Could you split this up in two;

* Add the new field and store/update it during POST/PATCH

Then we can run a cluster wide migration to get the field populated, then in the next step;

* Remove the call to Auth on PATCH/DELETE

**Comment by surajssd** [comment]
Once we have done migrations I will undo this commit https://github.com/fabric8-services/fabric8-tenant/pull/597/commits/6a4813157b52ce0fd4244f816a18253e5f3a27eb in a new PR!

**Comment by fabric8cd** [comment]
@surajssd snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-597-6`

**Comment by fabric8cd** [comment]
@surajssd snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-597-7`

File: controller/tenant.go [hunk 2]
```diff
@@ -177,6 +181,19 @@ func (c *TenantController) Update(ctx *app.UpdateTenantContext) error {
 	// create openshift config
 	openshiftConfig := openshift.NewConfig(c.defaultOpenshiftConfig, user, cluster.User, cluster.Token, cluster.APIURL)
 
+	// update tenant config
+	tenant := &tenant.Tenant{
```
Comment (aslakknutsen) [inline controller/tenant.go]: Why not update and save the Tenant object we already fetched on line 142?

**Comment by fabric8cd** [comment]
@surajssd snapshot fabric8-tenant image is available for testing.  `docker pull docker.io/fabric8/fabric8-tenant:SNAPSHOT-PR-597-8`

**Pull request merged**